src/datadoc/frontend/callbacks/register_callbacks.py

```diff
@@ -92,20 +95,37 @@ def callback_update_progress_label(
         State("alerts-section", "children"),
         prevent_initial_call=True,
     )
-    def callback_save_metadata_file(n_clicks: int, alerts: list) -> list:
-        """Save the metadata document to disk and check obligatory metadata."""
-        if alerts:
-            alerts = []
+    def callback_save_metadata_file(
+        n_clicks: int,
+        alerts: list,  # argument required by Dash  # noqa: ARG001
+    ) -> list:
+        """Save the metadata document to disk and check obligatory metadata.
+
+        Returns:
+            List of alerts. Obligatory metadata alert warning if there is obligatory metadata missing.
+            And success alert if metadata is saved correctly.
+            If none return no_update.
+        """
+        missing_obligatory_dataset = ""
+        missing_obligatory_variables = ""
         if n_clicks and n_clicks > 0:
-            state.metadata.write_metadata_document()
-            save = build_ssb_alert(
-                AlertTypes.SUCCESS,
-                "Lagret metadata",
-            )
+            with warnings.catch_warnings(record=True) as w:
+                warnings.simplefilter("always")
+                state.metadata.write_metadata_document()
+                save = build_ssb_alert(
+                    AlertTypes.SUCCESS,
+                    "Lagret metadata",
+                )
+                for warning in w:
+                    if warning is not None:
+                        if issubclass(warning.category, ObligatoryDatasetWarning):
+                            missing_obligatory_dataset = str(warning.message)
+                        elif issubclass(warning.category, ObligatoryVariableWarning):
```

Comment:
Did you consider having an else clause with logging in it? Otherwise branch misses will go past silently.

Reply:
I will look into it